pkg/infrastructure/infinispan_integration.go

```diff
@@ -123,8 +124,19 @@ func SetInfinispanVariables(infinispanProps v1alpha1.InfinispanConnectionPropert
 		len(infinispanProps.Credentials.SecretName) > 0 &&
 		secret != nil &&
 		container != nil {
-		framework.SetEnvVarFromSecret(infinispanEnvKeyUsername, InfinispanSecretUsernameKey, secret, container)
-		framework.SetEnvVarFromSecret(infinispanEnvKeyPassword, InfinispanSecretPasswordKey, secret, container)
+
+		if len(infinispanProps.Credentials.UsernameKey) > 0 {
+			framework.SetEnvVarFromSecret(infinispanEnvKeyUsername, infinispanProps.Credentials.UsernameKey, secret, container)
+		} else {
+			framework.SetEnvVarFromSecret(infinispanEnvKeyUsername, InfinispanSecretUsernameKey, secret, container)
+		}
```

Comment:
Since it only changes the second argument, I would write it this way:

```go
usernameValue := InfinispanSecretUsernameKey
if len(infinispanProps.Credentials.UsernameKey) > 0 {
    usernameValue = infinispanProps.Credentials.UsernameKey
}

framework.SetEnvVarFromSecret(infinispanEnvKeyUsername, usernameValue, secret, container)
```

The same with the below password key. What do you think about?

Reply:
Good point, will update.